src/Http/Routing/src/Patterns/RoutePatternFactory.cs

```diff
@@ -661,6 +624,52 @@ RoutePatternPart VisitPart(RoutePatternPart part)
         }
     }
 
+    /// <summary>
+    /// String policy references are infered to be regex constraints. Creating them is moved here to its own method so apps can
+    /// trim away the regex dependency when RoutePatternFactory.Parse(string) is used. This is the method typically used by the various Map methods.
+    /// </summary>
+    private static Dictionary<string, List<RoutePatternParameterPolicyReference>>? CreateRoutePatternPolicyReferences(RouteValueDictionary? parameterPolicies)
+    {
+        Dictionary<string, List<RoutePatternParameterPolicyReference>>? updatedParameterPolicies = null;
+        if (parameterPolicies != null && parameterPolicies.Count > 0)
+        {
+            updatedParameterPolicies = new Dictionary<string, List<RoutePatternParameterPolicyReference>>(parameterPolicies.Count, StringComparer.OrdinalIgnoreCase);
+
+            foreach (var kvp in parameterPolicies)
+            {
+                var policyReferences = new List<RoutePatternParameterPolicyReference>();
+
+                if (kvp.Value is IParameterPolicy parameterPolicy)
+                {
+                    policyReferences.Add(ParameterPolicy(parameterPolicy));
+                }
+                else if (kvp.Value is string)
+                {
+                    // Constraint will convert string values into regex constraints
+                    policyReferences.Add(Constraint(kvp.Value));
```

Comment:
Doesn't this still ultimately mean that ```new RegexRouteConstraint(...)``` could be called?

Reply:
... oh I think I see what you've done. You've pulled the problematic code path into CreateRoutePatternPolicyReferences and for the code path that we call by default in the minimal APIs scenario we don't actually call the overload of Parse that invokes it. So, we can rip that dependency away.